homeassistant/components/homeconnect/binary_sensor.py

```diff
@@ -0,0 +1,72 @@
+"""Provides a binary sensor for Home Connect.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/integrations/binary_sensor.homeconnect/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from .api import HomeConnectEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Home Connect binary sensor."""
+
+    def get_entities():
+        entities = []
+        hc_api = hass.data[DOMAIN][config_entry.entry_id]
+        for device_dict in hc_api.devices:
+            entity_dicts = device_dict.get("entities", {}).get("binary_sensor", [])
+            entity_list = [HomeConnectBinarySensor(**d) for d in entity_dicts]
+            device = device_dict["device"]
+            device.entities += entity_list
```

Comment:
Don't store the entities on the device.

Reply:
Oh right, this is no longer needed now that we're using the dispatcher.